Solution17.java

```diff
@@ -0,0 +1,30 @@
+public class Solution17 {
```

Comment:
correct but may exceed time limit for some edge cases. It takes a lot of time to finish. Can you improve? maybe you should check discussions in leetcode page of the problem